flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/AbstractStreamOperatorV2.java

```diff
@@ -508,4 +508,8 @@ public void setKeyContextElement2(StreamRecord<?> record) throws Exception {
 	protected Optional<InternalTimeServiceManager<?>> getTimeServiceManager() {
 		return Optional.ofNullable(timeServiceManager);
 	}
+
+	protected Optional<StreamOperatorStateHandler> getStateHandler() {
+		return Optional.ofNullable(stateHandler);
+	}
```

Comment:
In that case I would marked those methods as `@Internal` (they are both returning already `@Internal` marked class)